src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -2128,36 +2140,41 @@ def detach_managed_data_disk(cmd, resource_group_name, vm_name, disk_name=None,
 
 # region VirtualMachines Extensions
 def list_extensions(cmd, resource_group_name, vm_name):
-    vm = get_vm(cmd, resource_group_name, vm_name)
-    extension_type = 'Microsoft.Compute/virtualMachines/extensions'
-    result = [r for r in (vm.resources or []) if r.type == extension_type]
-    return result
```

Comment:
Doesn't the function get_vm() call the virtual machine get method? Why are we instead calling the VMExtensionList?